pkg/frontend/querymiddleware/split_by_instant_interval_test.go

```diff
@@ -23,279 +23,290 @@ import (
 )
 
 func TestQuerySplittingCorrectness(t *testing.T) {
-	var (
-		numSeries          = 1000
-		numStaleSeries     = 100
-		numHistograms      = 1000
-		numStaleHistograms = 100
-		histogramBuckets   = []float64{1.0, 2.0, 4.0, 10.0, 100.0, math.Inf(1)}
-	)
+	for _, startString := range []string{
+		"2020-01-01T03:00:00.100Z",
+		"2020-01-01T03:00:00Z",
+	} {
+		t.Run(fmt.Sprintf("start=%s", startString), func(t *testing.T) {
+			start, err := time.Parse(time.RFC3339Nano, startString)
+			require.NoError(t, err)
+			end := start.Add(30 * time.Minute)
```

Comment:
Given this doesn't affect the test cases definition, I would consider moving the for loop below, after tests cases are defined.